### ISSUE TITLE
Plugin should be enabled by node type

### ISSUE THREAD

**Issue by brollb** [webgme/webgme#720]
Plugins should be able to be enabled only for specific node types.

This would prevent the boilerplate code in the plugin which checks the node type and alerts the user when it is not allowed.

**Comment by rlhelinski** [comment]
I this supported in WebGME 2.30? How would one enable their own plugin for a meta type? Can this be done in the `metadata.json`? I only see the option `config.plugin.displayAll = false` in the documentation which seems like it enables the plugin for every type.

**Comment by rlhelinski** [comment]
@pmeijer Could you help?

**Comment by kecso** [comment]
The validPlugins can be set via the property editor. It is a multi-select field. Only plugins that are allowed for the specific node can be executed on that given node.

**Comment by rlhelinski** [comment]
@kecso Right. I have written a plug-in and want to somehow specify the type(s) of objects that are allowed so that the user does not have to click the check box in the Property Editor / Meta / validPlugins. Is that possible?

**Comment by kecso** [comment]
It is, but it is usually done from a different angle.
What you usually do is prepare a seed for your users. That seed project will contain the domain's meta-model where you can allow the given plugin for the proper concepts. Once it is done, your users will create their projects starting from that seed where this configuration is already done. So they are not required to do any such plugin configuration. (And usually, you package the plugins and seeds together in a github repository)
I hope it is clear and fits your scenario.
Configuration wise, you cannot really lock 'types/classes' to the plugin as there is no proper identification for that purpose(by-name is not unique enough, by-id would be too specific from the plugin's point of view).

**Comment by rlhelinski** [comment]
OK, thanks, @kecso
